vpc/main.tf

```diff
@@ -0,0 +1,110 @@
+#Create vpc
+resource "aws_vpc" "practice_vpc" {
+    cidr_block = "10.0.0.0/16"
+
+    tags = {
+        Name = "practice_vpc"
+    }
+}
+
+#Create subnets
+resource "aws_subnet" "private_subnet" {
+    vpc_id = aws_vpc.practice_vpc.id
+    cidr_block =  "10.0.0.0/24"
+
+    tags = {
+        Name = "private_subnet",
+        vpc = "practice_vpc"
+    }
+}
+
+resource "aws_subnet" "public_subnet" {
+    vpc_id = aws_vpc.practice_vpc.id
+    cidr_block =  "10.0.1.0/24"
+
+    tags = {
+      Name = "public_subnet" 
+      vpc = "practice_vpc"
+    }
+}
+
+#Create internet gateway
+resource "aws_internet_gateway" "internet_gateway" {
+  vpc_id = aws_vpc.practice_vpc.id
+
+  tags = {
+    Name = "internet_gateway",
+    vpc = "practice_vpc"
+  }
+}
+
+#Create Elastic IP for NAT Gateway
+resource "aws_eip" "eip_nat_gateway" {
+  domain = "vpc"
+  depends_on = [aws_internet_gateway.internet_gateway]
+  tags = {
+    Name = "nat_gateway_elastic_ip"
+    vpc = "practice_vpc"
+  }
+}
+
+#Create public NAT gateway
+resource "aws_nat_gateway" "nat_gateway" {
+  subnet_id     = aws_subnet.public_subnet.id
+  allocation_id = aws_eip.eip_nat_gateway.id
+
+  # To ensure proper ordering, it is recommended to add an explicit dependency
+  # on the Internet Gateway for the VPC.
+  depends_on = [aws_internet_gateway.internet_gateway]
+
+  tags = {
+    Name = "nat_gateway",
+    subnet = "public_subnet",
+    vpc = "practice_vpc"
+  }
+}
+
+#Create route tables
+resource "aws_route_table" "public_route_table" {
+  vpc_id = aws_vpc.practice_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0" 
+    gateway_id = aws_internet_gateway.internet_gateway.id
+  }
+
+  tags = {
```

Comment:
Tagging is very Critical when you are tracking cost on the cloud, reason why we put in some useful tags that will allow you to know where cost are coming from.
Owner , Environment, service that will use the resource is important to be added to the tags

Reply:
I understand, I will edit the tags to this instead.